src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 0);
+		add(blocks[2][1], gridBagConstraints);
+
+		blocks[2][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		add(blocks[2][2], gridBagConstraints);
+	}
+
+	public void populatePanel(GameSession gameSession, boolean isRefresh) {
+
+		this.gameSession = gameSession;
+		this.selectedCell = null;
+
+		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][1].populate(gameSession.getGameBoard().getBlock(1),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[0][2].populate(gameSession.getGameBoard().getBlock(2),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[1][0].populate(gameSession.getGameBoard().getBlock(3),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][1].populate(gameSession.getGameBoard().getBlock(4),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[1][2].populate(gameSession.getGameBoard().getBlock(5),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+
+		blocks[2][0].populate(gameSession.getGameBoard().getBlock(6),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][1].populate(gameSession.getGameBoard().getBlock(7),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+		blocks[2][2].populate(gameSession.getGameBoard().getBlock(8),
+				gameSession, isRefresh, buildFocusAdapter(), buildMouseAdapter());
+	}
+	
+	public void populatePencilMark(GameSession gameSession) {
+
+		this.gameSession = gameSession;
+
+		blocks[0][0].populatePencilMark(gameSession.getGameBoard().getBlock(0),
+				gameSession);
+		blocks[0][1].populatePencilMark(gameSession.getGameBoard().getBlock(1),
+				gameSession);
+		blocks[0][2].populatePencilMark(gameSession.getGameBoard().getBlock(2),
+				gameSession);
+
+		blocks[1][0].populatePencilMark(gameSession.getGameBoard().getBlock(3),
+				gameSession);
+		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
+				gameSession);
+		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
+				gameSession);
+
+		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
+				gameSession);
+		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
+				gameSession);
+		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
+				gameSession);
+	}
+
+	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
+		
+		if (null != selectedCell) {

```

Comment:
If you are referring (null != selectCell), that's a convention I learned
from my old tech-lead. This way, I know that particular condition is
written by me and not others. It's like a signature.



On Sun, Jul 12, 2015 at 10:11 PM, Jeremy Nelson <notifications@github.com>
wrote:

> In src/edu/psu/sweng500/team8/gui/BoardGUI.java
> <https://github.com/garnryang/Team8/pull/13#discussion_r34429612>:
>
> > +		blocks[1][1].populatePencilMark(gameSession.getGameBoard().getBlock(4),
> > +				gameSession);
> > +		blocks[1][2].populatePencilMark(gameSession.getGameBoard().getBlock(5),
> > +				gameSession);
> > +
> > +		blocks[2][0].populatePencilMark(gameSession.getGameBoard().getBlock(6),
> > +				gameSession);
> > +		blocks[2][1].populatePencilMark(gameSession.getGameBoard().getBlock(7),
> > +				gameSession);
> > +		blocks[2][2].populatePencilMark(gameSession.getGameBoard().getBlock(8),
> > +				gameSession);
> > +	}
> > +
> > +	public void updateSelectedCellFromHint(CellCoordinates cellCoordinates, int number) {
> > +		
> > +		if (null != selectedCell) {
>
> Same comment as earlier -- use this.selectedCell
>
> Also your Yoda-conditions make me LOL. Not that it matters -- I've just
> never seen anyone actually use them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/garnryang/Team8/pull/13/files#r34429612>.
>